src/server/game/Battlegrounds/BattlegroundMgr.cpp

```diff
@@ -821,7 +821,7 @@ void BattlegroundMgr::ToggleArenaTesting()
 
 void BattlegroundMgr::SetHolidayWeekends(uint32 mask)
 {
-    for (uint32 bgtype = 1; bgtype < MAX_BATTLEGROUND_TYPE_ID; ++bgtype)
+    for (uint64 bgtype = 1; bgtype < MAX_BATTLEGROUND_TYPE_ID; ++bgtype)
```

Comment:
Thsi won't fix the defect:

```
large_shift: In expression 1 << bgtype, left shifting by more than 31 bits has undefined behavior. The shift amount, bgtype, is as much as 32.
```

The actual issue here is that mask is shifted 32 bits (greatest value of bgtype) to the left which causes an overflow of 1 bit.
This isn't solved through increasing bgtype to 64 bit size.

![screenshot_1](https://cloud.githubusercontent.com/assets/1146834/9381891/f984cd0e-473b-11e5-9fff-897da0528ad1.png)